src/nodes/Networking/Inline.js

```diff
@@ -154,9 +154,23 @@ x3dom.registerNodeType(
             {
                 var that = this;
 
+		var isJSON = true;
+
                 var xhr = new window.XMLHttpRequest();
-                if (xhr.overrideMimeType)
-                    xhr.overrideMimeType('text/xml');   //application/xhtml+xml
+                if (this._vf.url.length && this._vf.url[0].length) {
+			if (this._vf.url[0].endsWith(".x3d")) {
+				isJSON = false;
+				if (xhr.overrideMimeType)
+				    xhr.overrideMimeType('text/xml');   //application/xhtml+xml
+			} else if (this._vf.url[0].endsWith(".json")) {
+				if (xhr.overrideMimeType)
+				    xhr.overrideMimeType('text/json');
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types suggests application/json which seems to be official mime type. Did text/json work better with xhr ?